cpp/src/parquet/encoding.cc

```diff
@@ -2265,21 +2269,27 @@ class DeltaLengthByteArrayDecoder : public DecoderImpl,
   }
 
   int Decode(ByteArray* buffer, int max_values) override {
+    // Decode up to `max_values` strings into an internal buffer
+    // and reference them into `buffer`.
     max_values = std::min(max_values, num_valid_values_);
 
-    int64_t data_size = 0;
+    int32_t data_size = 0;
```

Comment:
it is not immediately clear to me why this must be less than int32_t?  I guess I need to reread the spec on this.

Reply:
Not in the spec, but simply the fact that `GetBatch` takes an `int` for the batch size.

Reply:
I think pages can only hold int32max values in them (row groups can hold more and apparently we have a bug someplace when reading).  So this makes sense.